src/backend/plugins/organizations.py

```diff
@@ -81,3 +81,198 @@ def run_after_create(self, organization):
         organization.name = name
         organization.save(update_fields=["name", "updated_at"])
         logger.info("Organization %s name updated to %s", organization, name)
+
+    def run_after_grant_access(self, organization_access):
+        """After granting an organization access, we don't need to do anything."""
+
+
+class ApiCall:
+    """Encapsulates a call to an external API"""
+
+    inputs: dict = {}
+    method: str = "GET"
+    base: str = ""
+    url: str = ""
+    params: dict = {}
+    headers: dict = {}
+    response_data = None
+
+    def execute(self):
+        """Call the specified API endpoint with supplied parameters and record response"""
+        if self.method in ("POST", "PATCH"):
+            response = requests.request(
+                method=self.method,
+                url=f"{self.base}/{self.url}",
+                json=self.params,
+                headers=self.headers,
+                timeout=5,
+            )
+        else:
+            response = requests.request(
+                method=self.method,
+                url=f"{self.base}/{self.url}",
+                params=self.params,
+                headers=self.headers,
+                timeout=5,
+            )
+        self.response_data = response.json()
+        logger.info(
+            "API call: %s %s %s %s",
+            self.method,
+            self.url,
+            self.params,
+            self.response_data,
+        )
+
+
+class CommuneCreation(BaseOrganizationPlugin):
```

Comment:
As this is French-specific, this code probably shouldn't be in a `plugins/organizations.py` file ? Should we have a "fr" folder there?

Reply:
Similar reasoning to the above: my intent was to extend the existing plugin mechanism and not break new ground.